internal/services/netapp/netapp_volume_resource.go

```diff
@@ -391,8 +391,33 @@ func resourceNetAppVolume() *pluginsdk.Resource {
 				Default:     false,
 				Description: "Enable access based enumeration setting for SMB/Dual Protocol volume. When enabled, users who do not have permission to access a shared folder or file underneath it, do not see that shared resource displayed in their environment.",
 			},
+
+			"is_large_volume": {
+				Type:        pluginsdk.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Indicates whether the volume is a large volume.",
+			},
 		},
 		CustomizeDiff: func(ctx context.Context, d *pluginsdk.ResourceDiff, i interface{}) error {
+			// Validate large volume and storage_quota_in_gb based on Azure NetApp Files requirements
+			isLargeVolume := d.Get("is_large_volume").(bool)
+			storageQuotaInGB := d.Get("storage_quota_in_gb").(int)
+
+			switch {
+			case isLargeVolume && storageQuotaInGB < 51200:
+				// Large volumes must be at least 50 TiB (51,200 GB)
+				return fmt.Errorf("when is_large_volume is true, storage_quota_in_gb must be at least 51,200 GB (50 TiB)")
```

Comment:
Could we wrap the property names in backticks (`)?

```suggestion
				return fmt.Errorf("when `is_large_volume` is true, `storage_quota_in_gb` must be at least 51,200 GB (50 TiB)")
```